mypy/semanal.py

```diff
@@ -2128,6 +2128,17 @@ def visit_member_expr(self, expr: MemberExpr) -> None:
                 if full_name in obsolete_name_mapping:
                     self.fail("Module has no attribute %r (it's now called %r)" % (
                         expr.name, obsolete_name_mapping[full_name]), expr)
+        elif isinstance(base, RefExpr) and isinstance(base.node, TypeInfo):
```

Comment:
The previous block looks for foo.bar where foo is a module (and bar is anything defined in foo.py). The new block here looks for C.bar where C is some class (C could also be a more complex expression, like foo.C), but only of bar is a module object. For other types of things in C (e.g. methods or class variables) other existing code paths do the right thing, but those other paths (in particular in checkmember.py) don't get it right if bar happens to be a module. Without my small change there, the test code would crash (see original issue, #1839). But the code here (semanal.py) is required to create the right symbol table entry so that looking up things in C.bar will actually work.